include/CL/opencl.hpp

```diff
@@ -2396,35 +2385,39 @@ class Device : public detail::Wrapper<cl_device_id>
                 cl_uint * /*num_devices*/ ) CL_API_SUFFIX__VERSION_1_1;
 
         static PFN_clCreateSubDevicesEXT pfn_clCreateSubDevicesEXT = NULL;
-        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCreateSubDevicesEXT);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(platform, clCreateSubDevicesEXT);
 
-        cl_uint n = 0;
-        cl_int err = pfn_clCreateSubDevicesEXT(object_, properties, 0, NULL, &n);
-        if (err != CL_SUCCESS) {
-            return detail::errHandler(err, __CREATE_SUB_DEVICES_ERR);
-        }
+        if (pfn_clCreateSubDevicesEXT) {
+            cl_uint n = 0;
+            error = pfn_clCreateSubDevicesEXT(object_, properties, 0, NULL, &n);
+            if (error != CL_SUCCESS) {
+                return detail::errHandler(error, __CREATE_SUB_DEVICES_ERR);
+            }
 
-        vector<cl_device_id> ids(n);
-        err = pfn_clCreateSubDevicesEXT(object_, properties, n, ids.data(), NULL);
-        if (err != CL_SUCCESS) {
-            return detail::errHandler(err, __CREATE_SUB_DEVICES_ERR);
-        }
-        // Cannot trivially assign because we need to capture intermediates 
-        // with safe construction
-        if (devices) {
-            devices->resize(ids.size());
+            vector<cl_device_id> ids(n);
+            error = pfn_clCreateSubDevicesEXT(object_, properties, n, ids.data(), NULL);
+            if (error != CL_SUCCESS) {
+                return detail::errHandler(error, __CREATE_SUB_DEVICES_ERR);
+            }
 
-            // Assign to param, constructing with retain behaviour
-            // to correctly capture each underlying CL object
-            for (size_type i = 0; i < ids.size(); i++) {
-                // We do not need to retain because this device is being created 
-                // by the runtime
-                (*devices)[i] = Device(ids[i], false);
+            // Cannot trivially assign because we need to capture intermediates
+            // with safe construction
+            if (devices) {
+                devices->resize(ids.size());
+
+                // Assign to param, constructing with retain behaviour
+                // to correctly capture each underlying CL object
+                for (size_type i = 0; i < ids.size(); i++) {
+                    // We do not need to retain because this device is being created 
+                    // by the runtime
+                    (*devices)[i] = Device(ids[i], false);
+                }
             }
```

Comment:
IMHO this function is unfortunate in multiple ways, some of which are breaking so I won't list. However:
- if the user doesn't supply a vector, then the entire function is meaningless and we should shortcircuit the entire thing and not leak subdevice handles.
- the SDK utility library introduced new error codes for cases when the wrapper itself is misused. This function really has no meaning when a `nullptr` is provided by the user and we should likely signal that. Whether we repurpose an error code from [`clCreateSubDevices`](https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clCreateSubDevices) or start making our own, it's up to us. Silently leaking resources seems wrong.

I'm fine deferring this into a separate issue.